src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseTypeDefinition.java

```diff
@@ -168,4 +174,27 @@ public boolean isCaseFieldACollection(String caseFieldId) {
     public Optional<CaseFieldDefinition> getCaseField(String caseFieldId) {
         return caseFieldDefinitions.stream().filter(caseField -> caseField.getId().equalsIgnoreCase(caseFieldId)).findFirst();
     }
+
+    @JsonIgnore
+    public Optional<CommonField> getCommonFieldByPath(String path) {
+        if (StringUtils.isBlank(path)) {
+            return Optional.empty();
+        }
+        List<String> pathElements = Arrays.stream(path.trim().split("\\.")).collect(toList());
+
+        Optional<CaseFieldDefinition> topLevelCaseField = getCaseField(pathElements.get(0));
+        if (topLevelCaseField.isPresent()) {
+            return topLevelCaseField.get().getComplexFieldNestedField(pathElements.stream().skip(1).collect(Collectors.joining(",")));
+        }
+
+        return Optional.empty();
+    }
+
+    @JsonIgnore
+    public Map<String, TextNode> getLabelsFromCaseFields() {
+        return getCaseFieldDefinitions()
+            .stream()
+            .filter(caseField -> LABEL.equals(caseField.getFieldTypeDefinition().getType()))
```

Comment:
temporary: why we getting only the field of type label here?

Reply:
Just to not break existing behaviour with generating the case response data (this was originally in `MergeDataToSearchResultOperation`). Unsure of the original reasoning behind it, however..